src/ray/core_worker/test/task_manager_test.cc

```diff
@@ -1205,899 +1229,1032 @@ TEST_F(TaskManagerLineageTest, TestResubmittedDynamicReturnsTaskFails) {
-  dynamic_return_id = ObjectID::FromIndex(spec.TaskId(), 3);
-  data = GenerateRandomBuffer();
-  req = GetIntermediateTaskReturn(
-      /*idx*/ 1,
-      /*finished*/ false,
-      generator_id,
-      /*dynamic_return_id*/ dynamic_return_id,
-      /*data*/ data,
-      /*set_in_plasma*/ false);
-  signal_called = false;
-  ASSERT_TRUE(manager_.HandleReportGeneratorItemReturns(
-      req,
-      /*execution_signal_callback*/ [&signal_called](Status status,
-                                                     int64_t num_objects_consumed) {
-        signal_called = true;
-        ASSERT_TRUE(status.ok());
-        ASSERT_EQ(num_objects_consumed, 1);
-      }));
-  ASSERT_FALSE(signal_called);
-
-  // Worker failure. New worker should start reporting the task.
-  auto error = rpc::ErrorType::WORKER_DIED;
-  ASSERT_TRUE(manager_.FailOrRetryPendingTask(spec.TaskId(), error));
-
-  // Two report will come again. The first one should reply immediately (because)
-  // it is already replied and the second one should be backpressured.
-  /// 1 generate, 0 consumed, 2 threshold -> should signal immediately.
-  dynamic_return_id = ObjectID::FromIndex(spec.TaskId(), 2);
-  req = GetIntermediateTaskReturn(
-      /*idx*/ 0,
-      /*finished*/ false,
-      generator_id,
-      /*dynamic_return_id*/ dynamic_return_id,
-      /*data*/ data,
-      /*set_in_plasma*/ false);
-  bool retry_signal_called = false;
-  ASSERT_FALSE(manager_.HandleReportGeneratorItemReturns(
-      req,
-      /*execution_signal_callback*/ [&retry_signal_called](Status status,
-                                                           int64_t num_objects_consumed) {
-        retry_signal_called = true;
-        ASSERT_TRUE(status.ok());
-        ASSERT_EQ(num_objects_consumed, 0);
-      }));
-  ASSERT_TRUE(retry_signal_called);
-
-  /// 2 generate, 0 consumed, 2 threshold -> backpressured
-  dynamic_return_id = ObjectID::FromIndex(spec.TaskId(), 3);
-  data = GenerateRandomBuffer();
-  req = GetIntermediateTaskReturn(
-      /*idx*/ 1,
-      /*finished*/ false,
-      generator_id,
-      /*dynamic_return_id*/ dynamic_return_id,
-      /*data*/ data,
-      /*set_in_plasma*/ false);
-  retry_signal_called = false;
-  ASSERT_FALSE(manager_.HandleReportGeneratorItemReturns(
-      req,
-      /*execution_signal_callback*/ [&retry_signal_called](Status status,
-                                                           int64_t num_objects_consumed) {
-        retry_signal_called = true;
-        ASSERT_TRUE(status.ok());
-        ASSERT_EQ(num_objects_consumed, 1);
-      }));
-  // Backpressured.
-  ASSERT_FALSE(retry_signal_called);
-
-  ObjectID obj_id;
-  // Read should signal both executor.
-  auto status = manager_.TryReadObjectRefStream(generator_id, &obj_id);
-  ASSERT_TRUE(signal_called);
-  ASSERT_TRUE(retry_signal_called);
-  CompletePendingStreamingTask(spec, caller_address, 2);
-}
+// TEST_F(TaskManagerTest, TestObjectRefStreamDeletedStreamIgnored) {
```

Comment:
Are these tests temporarily disabled?

Reply:
Yes, working on putting these back before we merge. I made some changes to TaskManager interface so these were failing to compile.

Please review even though premerge is failing! Thanks!